src/environment_setup/biopam.mli

```diff
@@ -37,20 +36,77 @@ type install_target = {
   (** Test to determine success of the install.
       Defaults to `test -e witness`. *)
   test :
-    (?host:KEDSL.Host.t -> string -> KEDSL.Command.t) option;
+    (host:KEDSL.Host.t -> string -> KEDSL.Command.t) option;
 
   (* Install dependencies. *)
   edges : KEDSL.workflow_edge list;
 
   (* Transform the install and init programs, e.g. it needs to be run in a
      specific environment. Defaults to a “no-op”. *)
-  init_environment : KEDSL.Program.t option;
+  init_environment : install_path: string -> KEDSL.Program.t;
+
+  (** Whether this package requires Conda packages. *)
+  requires_conda: bool;
+
+  (** Which opam-repository the tool should come from. *)
+  repository: [ `Biopam | `Opam | `Custom of string ];
+
+  (** Which compiler should be used to create the tool's own installation
+      opam-switch.  *)
+  compiler: string option;
 }
 
-(** Provde the specified (via install_target) tool.*)
-val provide : ?host:KEDSL.Host.t ->
+val install_target:
+  ?tool_type:tool_type ->
+  ?test:(host: KEDSL.Host.t -> string -> KEDSL.Command.t) ->
+  ?edges: KEDSL.workflow_edge list ->
+  ?init_environment:(install_path:string -> KEDSL.Program.t) ->
+  ?requires_conda:bool ->
+  witness:string ->
+  ?package:string ->
+  ?repository:[ `Biopam | `Custom of string | `Opam ] ->
+  ?compiler:string ->
+  Machine.Tool.Definition.t ->
+  install_target
+(** Create {!install_target} values.
+
+    - [tool_type]: the kind of tool being installed.
+    See {{type:tool_type}tool_type}.
+    Default: [`Application].
+    - [test]: test to determine success of the install.
+    Default: ["test -e <witness>"].
+    - [edges]: dependencies to install first.
+    - [init_environment]: transform the install and init programs, e.g. it
+    needs to be run in a specific environment. Defaults to a “no-op”.
+    - [witness]: filename (basename)
+    that is passed to test determine success and what is exported (usually the
```

Comment:
yes I copied from what was in the type definition before and tried to expand but it's still not clear. Will change `TODO`.